src/wallet/rpcdump.cpp

```diff
@@ -510,3 +534,110 @@ UniValue dumpwallet(const UniValue& params, bool fHelp)
+                                                                                                                                                                                                  "{ \"type\":\"pubkey\", \"value\":\"<my public key>\", \"label\":\"example 1\", \"timestamp\":1455191480 } ], true") +
+            HelpExampleRpc("importmulti", "'[{ \"type\":\"pubkey\", \"value\":\"<my public key>\", \"label\":\"example 1\", \"timestamp\":1455191464 } ]', false") +
+
+            "\nResponse is an array with the same size as the input that has the execution result :\n"
+            "  [ { \"result\": true | false } , ... ]\n");
+
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (params.size() == 1)
+        RPCTypeCheck(params, boost::assign::list_of(UniValue::VARR));
+    else
+        RPCTypeCheck(params, boost::assign::list_of(UniValue::VARR)(UniValue::VBOOL));
+
+    const UniValue& jRequest = params[0].get_array();
+    bool fRescan = params.size() > 1 ? params[1].get_bool() : true;
+
+    UniValue jResponse(UniValue::VARR);
+
+    //Locks the wallet
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+    EnsureWalletIsUnlocked();
+
+    bool fRunScan = false;
+    int64_t nCurrentTimestamp = 0;
+
+    //Import values
+    BOOST_FOREACH (const UniValue& importRequest, jRequest.getValues()) {
+        UniValue result = UniValue(UniValue::VOBJ);
+        try {
+            if (fRescan) {
+                //To check the minimum timestamp (if rescan)
+                if (importRequest.exists("timestamp")) {
+                    int64_t timestamp = find_value(importRequest, "timestamp").get_int64();
+                    if (nCurrentTimestamp == 0 || timestamp < nCurrentTimestamp)
+                        nCurrentTimestamp = timestamp;
+                }
+            }
+
+            const string& strType = find_value(importRequest, "type").get_str();
+            const string& strValue = find_value(importRequest, "value").get_str();
+            string strLabel = importRequest.exists("label") ? find_value(importRequest, "label").get_str() : "";
+
+            if (strType == "privkey")
+                ImportPrivateKey(strValue, strLabel);
+            else if (strType == "pubkey")
+                ImportPublicKey(strValue, strLabel);
+            else if (strType == "address") {
+                bool fP2SH = importRequest.exists("p2sh") ? find_value(importRequest, "p2sh").get_bool() : false;
+                ImportAddressKey(strValue, strLabel, fP2SH);
+            } else
+                throw;
+
+            //Success
+            result.pushKV("result", UniValue(true));
+            fRunScan = true;
+        } catch (...) {
+            //Failed
+            result.pushKV("result", UniValue(false));
+        }
+
+        jResponse.push_back(result);
+    }
+
+    if (fRescan && fRunScan && jRequest.size()) {
+        //For default is genesis
+        CBlockIndex* pindex = chainActive.Genesis();
```

Comment:
Could initialise to `chainActive.FindLatestBefore(nCurrentTimestamp)` even when `nCurrentTimestamp == 0` right?